RecoHI/HiEvtPlaneAlgos/src/HiEvtPlaneFlatProducer.cc

```diff
@@ -205,39 +289,43 @@ HiEvtPlaneFlatProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetu
   for(int i = 0; i<NumEPNames; i++) {
     ep[i]=0;
   }
-  for (EvtPlaneCollection::const_iterator rp = evtPlanes->begin();rp !=evtPlanes->end(); rp++) {
-    if(rp->angle() > -5) {
-      string baseName = rp->label();
-      for(int i = 0; i< NumEPNames; i++) {
-	if(EPNames[i].compare(baseName)==0) {
-	  double psiFlat = flat[i]->GetFlatPsi(rp->angle(),vzr_sell,bin);
-	  epang[i]=psiFlat;
-	  if(EPNames[i].compare(rp->label())==0) {	    
-	    if(storeNames_) ep[i]= new EvtPlane(psiFlat, rp->sumSin(), rp->sumCos(),rp->label().data());
-	    else ep[i]= new EvtPlane(psiFlat, rp->sumSin(), rp->sumCos(),"");
-	  } 
-	}
-      }
-    }    
+  int indx = 0;
+  for (EvtPlaneCollection::const_iterator rp = evtPlanes_->begin();rp !=evtPlanes_->end(); rp++) {
+	double angorig = rp->angle();
+	double s = rp->sumSin();
+	double c = rp->sumCos();
+	double w = rp->sumw();
+	uint m = rp->mult();
+
+	double psiOffset = angorig;
+
+	if(useOffsetPsi_) psiOffset = flat[indx]->OffsetPsi(s,c,w,m,vzr_sell,bin);
+	double psiFlat = flat[indx]->getFlatPsi(psiOffset,vzr_sell,bin);
```

Comment:
as suggested in the other place, use only a ``` const HiEvtPlaneFlatten* ``` part so that there is no (unwanted) information sharing between events